lib/SIL/SILModule.cpp

```diff
@@ -762,3 +763,30 @@ bool SILModule::isOptimizedOnoneSupportModule() const {
   return getOptions().Optimization >= SILOptions::SILOptMode::Optimize &&
          isOnoneSupportModule();
 }
+
+void SILModule::setSerializeSILAction(SILModule::ActionCallback Action) {
+  assert(!SerializeSILAction && "Serialization action can be set only once");
+  SerializeSILAction = Action;
+}
+
+SILModule::ActionCallback SILModule::getSerializeSILAction() const {
+  return SerializeSILAction;
+}
+
+/// Removes [serialized] from all functions. This allows for more
+/// optimizations and for a better dead function elimination.
+static void removeSerializedFlagFromAllFunctions(SILModule &M) {
```

Comment:
I would move this into the SIL pass; I would not expect SILModule::serialize() itself to have any side effects on the module's declarations, for example I might call it during debugging in the middle of the pipeline.

Reply:
OK. I'll move it back. It was there already ;-)

Reply:
Done